### ISSUE TITLE
cli: fmt's target extension must be '.d2' only

### ISSUE THREAD

**Issue by bo-ku-ra** [terrastruct/d2#1063]
text files would be corrupted.( *.bat *.txt and so on)

**Comment by alixander** [comment]
i'm not sure. `gofmt` doesn't validate extension. indeed it shouldn't corrupt though. it should just error if it's not legal D2.

**Comment by bo-ku-ra** [comment]
ex.
markdown_text.md (not d2_file)
```
foo  \n  <---------- foo\x20\x20\n
bar\n
```
 c:>d2 fmt markdown_text.md
```
foo\n  <----- corrupted!
bar\n
```

**Comment by bo-ku-ra** [comment]
i think d2 must not overwrite file. (except with force overwrite option)

c:>d2 ```input```.d2 ```output```.(svg|png|pdf)

but when using fmt 

c:>d2 ```fmt``` input&output-file1 input&output-file2 input&output-file3 ...

i think users get confused.

**Comment by bo-ku-ra** [comment]
ex)
c:>d2 fmt foobar.d2

foobar_org.d2 <- original foobar.d2
foobar.d2        <- formated

c:>d2 fmt ```-o``` foobar.d2
foobar.d2        <- formated

**Comment by bo-ku-ra** [comment]
d2 v0.5.0 defined by 'import' with file extension '.d2'. (e.g. @foo.d2 -> @foo)
this issue should also  be '.d2' only.

**Comment by alixander** [comment]
@bo-ku-ra i don't understand your last comment. can you rephrase?

**Comment by bo-ku-ra** [comment]
fmt's target extension must be '.d2' only

**Comment by alixander** [comment]
tested, it does not corrupt, just outputs errors, same behavior as `gofmt`
